src/observer/net/cli_communicator.cpp

```diff
@@ -67,10 +67,16 @@ char *my_readline(const char *prompt)
   fprintf(stdout, "%s", prompt);
   char *s = fgets(buffer, MAX_MEM_BUFFER_SIZE, stdin);
   if (nullptr == s) {
-    free(buffer);
     if (ferror(stdin) || feof(stdin)) {
       LOG_WARN("failed to read line: %s", strerror(errno));
     }
+    /* strerrno(4)=="Interrupted system call"*/
+    if (errno == 4) {
```

Comment:
could you use errno name instead of number?

Reply:
Just like "if(0==strcmp(strerror(errno),"Interrupted system call"))"?

Reply:
No, like this:
```cpp
EINTR == errno
```